Vagrantfile

```diff
@@ -1,159 +1,92 @@
+      client.vm.network :private_network, ip: "192.168.58.100", netmask: "24"
+      client.vm.hostname = "client"
+      client.vm.provision "shell", path: "./scripts/init.sh"
+    end
+  
+    config.vm.define :server do |server|
+      server.vm.box = "fedora/38-cloud-base"
+      server.vm.box_version = "38.20230413.1"
+      server.vm.network :private_network, ip: "192.168.56.101", netmask: "24"
+      server.vm.network :private_network, ip: "192.168.57.101", netmask: "24"
+      server.vm.network :private_network, ip: "192.168.58.101", netmask: "24"
+      server.vm.hostname = "server"
+      server.vm.provision "shell", path: "./scripts/init.sh"
+    end
+  
+    # Disable automatic box update checking. If you disable this, then
+    # boxes will only be checked for updates when the user runs
+    # `vagrant box outdated`. This is not recommended.
+    # config.vm.box_check_update = false
+  
+    # Create a forwarded port mapping which allows access to a specific port
+    # within the machine from a port on the host machine. In the example below,
+    # accessing "localhost:8080" will access port 80 on the guest machine.
+    # NOTE: This will enable public access to the opened port
+    # config.vm.network "forwarded_port", guest: 80, host: 8080
+  
+    # Create a forwarded port mapping which allows access to a specific port
+    # within the machine from a port on the host machine and only allow access
+    # via 127.0.0.1 to disable public access
+    # config.vm.network "forwarded_port", guest: 80, host: 8080, host_ip: "127.0.0.1"
+  
+    # Create a private network, which allows host-only access to the machine
+    # using a specific IP.
+    # config.vm.network "private_network", ip: "192.168.33.10"
+  
+    # Create a public network, which generally matched to bridged network.
+    # Bridged networks make the machine appear as another physical device on
+    # your network.
+    # config.vm.network "public_network"
+  
+    # Share an additional folder to the guest VM. The first argument is
+    # the path on the host to the actual folder. The second argument is
+    # the path on the guest to mount the folder. And the optional third
+    # argument is a set of non-required options.
+    # config.vm.synced_folder "../data", "/vagrant_data"
+  
+    # Disable the default share of the current code directory. Doing this
+    # provides improved isolation between the vagrant box and your host
+    # by making sure your Vagrantfile isn't accessible to the vagrant box.
+    # If you use this you may want to enable additional shared subfolders as
+    # shown above.
+    # config.vm.synced_folder ".", "/vagrant", disabled: true
+  
+    # Provider-specific configuration so you can fine-tune various
+    # backing providers for Vagrant. These expose provider-specific options.
+    # Example for VirtualBox:
+    #
+    config.vm.provider "virtualbox" do |vb|
+        vb.memory = 4096
```

Comment:
(not sure we need that much: Maybe restrict to 2GB, not to require min > 8GB of RAM)